src/main/java/org/zalando/putittorest/HttpClientFactoryBean.java

```diff
@@ -39,6 +52,26 @@ public void setSocketTimeout(final int socketTimeout) {
         config.setSocketTimeout(socketTimeout);
     }
 
+    public void setTrustedKeystore(final Keystore keystore) throws Exception {
+        final SSLContextBuilder contextBuilder = SSLContexts.custom();
+
+        final String path = keystore.getPath();
+        final String password = keystore.getPassword();
+
+        final URL resource = HttpClientFactoryBean.class.getClassLoader().getResource(path);
+        if (resource == null) {
+            throw new FileNotFoundException(format("Keystore [%s] not found.", path));
+        }
+
+        try {
+            contextBuilder.loadTrustMaterial(resource, password == null ? null : password.toCharArray());
+            builder.setSSLSocketFactory(new SSLConnectionSocketFactory(contextBuilder.build(), getDefaultHostnameVerifier()));
```

Comment:
We may want to add a note somewhere that if you set this and use the `customizer` you get unexpected results. On the other hand: 
![With Great Power Comes Great Responsibility](http://quoteinvestigator.com/wp-content/uploads/2015/07/spider400.jpg)